src/components/HomeLandingPage/HomeLandingPage.tsx

```diff
@@ -40,24 +43,76 @@ const HomeLandingPage = () => {
   useAllEvents();
 
   // Fetch events from cache
-  const { data } = useQuery(GET_ALL_EVENTS);
+  const { data } = useQuery(FETCH_ALL_EVENTS);
+  const [editEvent] = useMutation(EDIT_EVENT);
+  const [deleteEvent] = useMutation(DELETE_EVENT, {
+    update(cache, { data: { deleteEvent } }) {
+      if (!deleteEvent) {
+        return;
+      }
+      let { events } = cache.readQuery<any>({
+        query: FETCH_ALL_EVENTS,
+      });
+
+      events = events.filter((event) => event.id !== eventToBeDeleted?.id);
+
+      cache.writeQuery({
+        query: FETCH_ALL_EVENTS,
+        data: { events },
+      });
+    },
+  });
   const events: Array<Event> = data ? data.events : [];
 
+  useEffect(() => {
+    if (eventToBeDeleted) {
+      deleteEvent({
+        variables: {
+          id: eventToBeDeleted?.id,
+        },
+      });
+    }
+  }, [eventToBeDeleted]);
```

Comment:
@JasonYG I still don't think that it's a good to do this in `useEffect`... I think `deleteEvent` should actually be called in `handleDeleteEvent`. You can set `eventToBeDeleted` when the user opens the "popper" with the options for the particular event card.

Reply:
Yeah I'd agree, having this in `useEffect` doesn't seem like the best way to do this, but setting `eventToBeDeleted` when the "popper" is opened requires passing the state setter down two components. I feel like the original solution I had with async/await is prob best here tbh